src/main/java/com/beust/jcommander/JCommander.java

```diff
@@ -735,6 +729,32 @@ private void parseValues(String[] args, boolean validate) {
         boolean isDashDash = false; // once we encounter --, everything goes into the main parameter
         while (i < args.length && !commandParsed) {
             String arg = args[i];
+
+            // 
+            // Expand @
+            // 
+            if (arg.startsWith("@") && options.expandAtSign) {
+                String fileName = arg.substring(1);
+                List<String> fileArgs = readFile(fileName);
+                
+                // Create a new array to hold the expanded arguments
+                String[] newArgs = new String[args.length + fileArgs.size() - 1];
+
+                // Copy the existing arguments before the '@' argument
+                System.arraycopy(args, 0, newArgs, 0, i);
+
+                // Copy the arguments from the file
+                for (int j = 0; j < fileArgs.size(); j++) {
```

Comment:
I don't think that performance plays a measurable role here, so we could simplify using `System.arraycopy(fileArgs.toArray(), 0, newArgs, i, fileArgs.length());` if you like. 😃

Reply:
I did that in 06e94745ca9d3321fef52a5b454324da84224bf9, and I enhanced the test in  842481f3dc40c3e1208748c54efaee6568b5397d to cover more cases.